simple-post/frontend/src/components/Posts.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component, Fragment } from 'react';
+import { get } from '../lib/API';
+
+import Post from './Post';
+
+class Posts extends Component {
+  state = {
+    posts: [],
+  };
+
+  componentDidMount = () => {
+    let pageNumber = localStorage.getItem('post-pageNumber');
+    if (!pageNumber) {
+      pageNumber = 1;
+      localStorage.setItem('post-pageNumber', pageNumber);
+    }
+    this.loadPosts(pageNumber);
+  }
+
+  loadPosts = async (pageNumber) => {
+    const postsResult = await get(`/posts/${pageNumber}`);
+    if (!postsResult.data) {
+      // handle no data here.
+    } else {
+        this.setState({
+            posts: postsResult.data,
+          });
```

Comment:
Do you think that handling the truthy condition in the `if()` itself would suffice, instead of leaving it to come to else on failing the condition in `if`? Would love to know your thoughts on this.